From f60e000628aea16c9b805baa1df5d5048b3ceaea Mon Sep 17 00:00:00 2001
From: Roland Haas <roland.haas@physics.gatech.edu>
Date: Wed, 2 Mar 2011 09:34:55 -0500
Subject: [PATCH] CarpetLibs: add missing (but harmless) omp private clause

  tmp1,tmp2 were missing, but since this code was never parallel executed by
  OpenMP (since we are already called by the max number of threads in
  data.cc/call_operator) it did not do any harm
---
 Carpet/CarpetLib/src/prolongate_3d_real8_eno.F90  |    2 +-
 Carpet/CarpetLib/src/prolongate_3d_real8_weno.F90 |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Carpet/CarpetLib/src/prolongate_3d_real8_eno.F90 b/Carpet/CarpetLib/src/prolongate_3d_real8_eno.F90
index 4b5be09..c70c124 100644
--- a/Carpet/CarpetLib/src/prolongate_3d_real8_eno.F90
+++ b/Carpet/CarpetLib/src/prolongate_3d_real8_eno.F90
@@ -194,7 +194,7 @@ subroutine prolongate_3d_real8_eno (src, srciext, srcjext, &
 
 !!$     Loop over fine region
 
-!$omp parallel do private (k0, fk, j0, fj, i0, fi)
+!$omp parallel do private (k0, fk, j0, fj, i0, fi, tmp1, tmp2)
   do k = 0, regkext-1
     k0 = (srckoff + k) / dstkfac
     fk = mod(srckoff + k, dstkfac)
diff --git a/Carpet/CarpetLib/src/prolongate_3d_real8_weno.F90 b/Carpet/CarpetLib/src/prolongate_3d_real8_weno.F90
index 199f455..e1844f6 100644
--- a/Carpet/CarpetLib/src/prolongate_3d_real8_weno.F90
+++ b/Carpet/CarpetLib/src/prolongate_3d_real8_weno.F90
@@ -271,7 +271,7 @@ subroutine prolongate_3d_real8_weno (src, srciext, srcjext, &
 
 !!$     Loop over fine region
 
-!$omp parallel do private (k0, fk, j0, fj, i0, fi)
+!$omp parallel do private (k0, fk, j0, fj, i0, fi, tmp1, tmp2)
   do k = 0, regkext-1
     k0 = (srckoff + k) / dstkfac
     fk = mod(srckoff + k, dstkfac)
-- 
1.7.1

